### ISSUE TITLE
Support for labels

### ISSUE THREAD

**Issue by MunsterPlop** [Mishkun/ideavim-sneak#16]
Greetings,

On vim, I make extensive use of

`let g:sneak#label = 1`

From vim-sneak's help pages:

> Label-mode minimizes the steps to jump to a location, using a clever interface
similar to EasyMotion. If enabled, Sneak overlays text with "labels" which can be jumped-to by typing the label character. 

I think it would be nice to have it as an option in Ideavim as well.

**Comment by Mishkun** [comment]
Greetings, @MunsterPlop!
Thanks for submitting this issue. Dont you think that this functionality heavily overlaps with cool [AceJump](https://github.com/acejump/AceJump) plugin?

**Comment by MunsterPlop** [comment]
@Mishkun 

I checked AceJump and it's pretty good but it's either one character target or incremental targeting.
I much prefer the two characters targeting of sneak as well as the fact that it jumps to the closest match before asking you for labels and one last thing I like is being able to jump to the next match through repeat (; and ,).

**Comment by Mishkun** [comment]
Okay, I understand your point. This functionality is lowest priority for me, but I would accept pull requests regarding this feature.

**Comment by MunsterPlop** [comment]
Maybe using AceJump's features to implement this like ideavim-easymotion does would be most convenient, though I suppose rolling a home-made solution makes more sense to keep this plugin simple/minimal like the original sneak.
I might look into this if I get the time, no promises though.

**Comment by colossatr0n** [comment]
I've added the necessary actions to AceJump to start emulating vim-sneak's label behavior. See this commit https://github.com/acejump/AceJump/commit/54ea6ac9c1b70499cb17725e11cbe14420d984b1. 

It allows AceJump's jump mode to be activated below or above the cursor.

Also, AceJump has a setting that allows for specifying the number of query characters. So if you set that to 2, and use the actions in the commit above, it gets close to how labels work in vim-sneak.

 I'll try to post an update here when the new AceJump version is released.

**Comment by colossatr0n** [comment]
Since it's been a while, I've opened up an issue in the AceJump repo to get the PR released. It's been merged to master, but it hasn't been released yet. See https://github.com/acejump/AceJump/issues/408 if you want to track the status.

**Comment by colossatr0n** [comment]
It’s now possible to emulate vim-sneak’s label behavior purely through AceJump. The changes were released with AceJump plugin version [3.8.8](https://plugins.jetbrains.com/plugin/7086-acejump/versions/stable/189931).

To do so:

1. Download AceJump
2. Add the following to your `.ideavimrc` (you may need to disable ideavim-sneak)
```
" Press `S` in normal mode to activate AceJump mode before the caret
nmap S <Action>(AceBackwardAction)

" Press `s` in normal mode to activate AceJump mode after the caret
nmap s <Action>(AceForwardAction)
```
3. Go to the AceJump settings and set "Minimum typed characters" to 2.

I'm pretty sure these configuration steps could be automatically added to ideavim-sneak in the future through something like `let g:sneak#label = 1`, but for now, it has to be done outside of ideavim-sneak.

**Comment by Mishkun** [comment]
> It’s now possible to emulate vim-sneak’s label behavior purely through AceJump. The changes were released with AceJump plugin version [3.8.8](https://plugins.jetbrains.com/plugin/7086-acejump/versions/stable/189931).

I think that closes the issue. Thanks!
